### PR TITLE
HDF5

### PR DESCRIPTION
**Pull request by cryos**

### CODE REVIEW

File: parsers/h5_parser.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  def uuid(self) -> UUID:
+    return UUID('{7420a76b-f0de-4fb1-a932-b6cb969f7af6}')
+
+  def supported_file_extensions(self) -> list:
+    return self.ext_list
+  
+  def accepts_extension(self, extension: str) -> bool:
+    if extension in self.ext_list:
+      return True
+    return False
```
Comment (joeykleingers) [inline parsers/h5_parser.py]:
```suggestion
    return extension in self.ext_list
```

**Review by joeykleingers — CHANGES_REQUESTED** [review]

File: parsers/h5_parser.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+  def type_dispatch(self, value: any) -> str|None:
```
Comment (joeykleingers) [inline parsers/h5_parser.py]:
```suggestion
  def type_dispatch(self, value: any) -> str:
```

File: parsers/h5_parser.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    elif isinstance(value, (numpy.ndarray, numpy.generic)):
+      if value.shape == (1,):
+        return str(value[0])
+      else:
+        return str(value)
+    
+    print(f'value confused us: {value} {type(value)}')
+    return None
+
+  def unpack_array(self, value: h5py.Dataset) -> str|None:
```
Comment (joeykleingers) [inline parsers/h5_parser.py]:
```suggestion
  def unpack_array(self, value: h5py.Dataset) -> str:
```

**Review by joeykleingers — CHANGES_REQUESTED** [review]

File: parsers/h5_parser.py (hunk 2 above)
Comment (joeykleingers) [reply parsers/h5_parser.py]: This pipe character in the return value errors out on my machine.

File: parsers/h5_parser.py (hunk 3 above)
Comment (joeykleingers) [reply parsers/h5_parser.py]: This pipe character in the return value errors out on my machine.

**Comment by cryos** [comment]
@joeykleingers seemed reasonable, local testing everything looks OK. I think this is ready to merge

**Review by joeykleingers — APPROVED** [review]

**Pull request merged**